test/extended/networking/egressip_helpers.go

```diff
@@ -0,0 +1,1330 @@
+)
+
+const (
+	nodeEgressIPConfigAnnotationKey = "cloud.network.openshift.io/egress-ipconfig"
+	// egressIPAgnImage                = "k8s.gcr.io/e2e-test-images/agnhost:2.36" // TODO: remove
+	// tcpdumpImage = "registry.ci.openshift.org/ocp/4.11:network-tools"
+)
+
+// TODO: make port allocator a singleton, shared among all test processes for egressip
+// TODO: add an egressip allocator similar to the port allocator
+
+type byName []corev1.Node
+
+func (n byName) Len() int           { return len(n) }
+func (n byName) Swap(i, j int)      { n[i], n[j] = n[j], n[i] }
+func (n byName) Less(i, j int) bool { return n[i].Name < n[j].Name }
+
+// GetNodesOrdered returns a sorted slice (by node.Name) of nodes or error.
+func getWorkerNodesOrdered(clientset kubernetes.Interface) ([]corev1.Node, error) {
+	if clientset == nil {
+		return nil, fmt.Errorf("Nil pointer clientset provided")
+	}
+
+	nodes, err := clientset.CoreV1().Nodes().List(
+		context.TODO(),
+		metav1.ListOptions{
+			LabelSelector: "node-role.kubernetes.io/worker=",
+		})
+	if err != nil {
+		return nil, err
+	}
+	items := nodes.Items
+	sort.Sort(byName(items))
+
+	return items, nil
+}
+
+// createPacketSnifferDaemonSet creates packet sniffer pods on the hosts specified in scheduleOnHosts.
+func createPacketSnifferDaemonSet(oc *exutil.CLI, namespace string, scheduleOnHosts []string, packetCaptureProtocol string, packetCapturePort int) (*appsv1.DaemonSet, error) {
+	f := oc.KubeFramework()
+	clientset := f.ClientSet
+
+	tcpdumpImage, err := DetermineImageFromRelease(oc, "network-tools")
+	if err != nil {
+		return nil, err
+	}
+
+	daemonsetName := fmt.Sprintf("%s-packet-sniffer", namespace)
+	targetDaemonset, err := createHostNetworkedPacketSnifferDaemonSet(
+		clientset,
+		tcpdumpImage,
+		packetCaptureProtocol,
+		packetCapturePort,
+		namespace,
+		daemonsetName,
+		scheduleOnHosts,
+	)
+	if err != nil {
+		return targetDaemonset, err
+	}
+
+	var ds *appsv1.DaemonSet
+	retries := 12
+	pollInterval := 5
+	for i := 0; i < retries; i++ {
+		// Get the DS
+		ds, err = clientset.AppsV1().DaemonSets(namespace).Get(context.TODO(), daemonsetName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		// Check if NumberReady == DesiredNumberScheduled.
+		// In that case, simply return as all went well.
+		if ds.Status.NumberReady == ds.Status.DesiredNumberScheduled {
+			return ds, nil
+		}
+		// If no port conflict error was found, simply sleep for pollInterval and then
+		// check again.
+		time.Sleep(time.Duration(pollInterval) * time.Second)
+	}
+
+	// The DaemonSet is not ready, but this is not because of a port conflict.
+	// This shouldn't happen and other parts of the code will likely report this error
+	// as a CI failure.
+	return ds, fmt.Errorf("Daemonset still not ready after %d tries", retries)
+}
+
+const (
+	tcpCaptureScript = `#!/bin/bash
+tcpdump -nne -i any -l -s 0  'port %d and tcp[((tcp[12:1] & 0xf0) >> 2):4] = 0x47455420' | awk '{print $9 " " $(NF-1)}'
```

Comment:
I know it would require a bigger refactor, but since it popped in to my head I figured to note it. I remember writing tests using tcpdump and using the -w <filename> option (as a background process), then doing a 'killall tcpdump' to finish it when it should be done writing. then tcpdump -r <filename> to read the packets back and parse what I needed.
That helped with some of the weirdness I would see when not actually capturing packets it should have.

BUT, looks like you have some good methods here as well.